### PR TITLE
Only define a single OID for composite objects

### LINKED ISSUE #50: Only define a single OID for composite algorithms
I think the `id-CompositeSignature` and `id-CompositePrivateKey` OIDs aren't necessary.

Signature OIDs only need to be defined when combined with a pre-hash algorithm (e.g. `ecdsa-with-sha384`), but a composite signature isn't combined with a pre-hash algorithm but rather each component signature may use a different (or no) pre-hash algorithm (e.g. ecdsa-with-sha384 + HSS).  EdDSA and HSS don't define different OIDs for the signature algorithm for the same reason that they don't have a pre-hash applied before the signature.

I'm not aware of any algorithms where a separate OID is defined for a private key version of the algorithm, it just uses the same public key OID (in fact I just added a definition of `CompositePrivateKey` used in `pk-CompositePublicKey` defined above which I think covers defining a composite private key using a single composite public key OID).

If we look at two of the more recent RFCs or drafts to define new algorithms for use in PKIX-related standards we have:
https://tools.ietf.org/html/draft-ietf-lamps-cms-hash-sig-08
```
   --
   -- Object Identifiers
   --
   id-alg-hss-lms-hashsig OBJECT IDENTIFIER ::= { iso(1)
       member-body(2) us(840) rsadsi(113549) pkcs(1) pkcs9(9)
       smime(16) alg(3) 17 }

   --
   -- Signature Algorithm and Public Key
   --

   sa-HSS-LMS-HashSig SIGNATURE-ALGORITHM ::= {
       IDENTIFIER id-alg-hss-lms-hashsig
       PARAMS ARE absent
       PUBLIC-KEYS { pk-HSS-LMS-HashSig }
       SMIME-CAPS { IDENTIFIED BY id-alg-hss-lms-hashsig } }

   pk-HSS-LMS-HashSig PUBLIC-KEY ::= {
       IDENTIFIER id-alg-hss-lms-hashsig
       KEY HSS-LMS-HashSig-PublicKey
       PARAMS ARE absent
       CERT-KEY-USAGE
           { digitalSignature, nonRepudiation, keyCertSign, cRLSign } }
```

and

https://tools.ietf.org/html/rfc8410
```
   id-X25519        OBJECT IDENTIFIER ::= { id-edwards-curve-algs 110 }
   id-Ed25519       OBJECT IDENTIFIER ::= { id-edwards-curve-algs 112 }

    sa-Ed25519 SIGNATURE-ALGORITHM ::= {
       IDENTIFIER id-Ed25519
        PARAMS ARE absent
        PUBLIC-KEYS {pk-Ed25519}
        SMIME-CAPS { IDENTIFIED BY id-Ed25519 }
    }

    pk-Ed25519 PUBLIC-KEY ::= {
        IDENTIFIER id-Ed25519
        -- KEY no ASN.1 wrapping --
        PARAMS ARE absent
        CERT-KEY-USAGE {digitalSignature, nonRepudiation,
                        keyCertSign, cRLSign}
        PRIVATE-KEY CurvePrivateKey
    }
```

So for the OIDs I'd pick `id-CompositePublicKey` (or `id-alg-compositePublicKey` if you want to be more like HSS).

And I think our draft's definition of composite Signature Algorithm and Public Key corresponding to RFC 5911 would be:
```
   sa-CompositeSignature SIGNATURE-ALGORITHM ::= {
       IDENTIFIER id-CompositePublicKey
       VALUE CompositeSignatureValue
       PARAMS TYPE CompositeSignatureAlgorithmParams ARE required
       PUBLIC-KEYS { pk-CompositePublicKey }
       SMIME-CAPS { IDENTIFIED BY id-CompositePublicKey } }

CompositeSignatureAlgorithmParams ::= SEQUENCE SIZE (1..MAX) OF AlgorithmIdentifier

CompositeSignatureValue ::= SEQUENCE SIZE (1..MAX) OF BIT STRING

   pk-CompositePublicKey PUBLIC-KEY ::= {
       IDENTIFIER id-CompositePublicKey
       KEY CompositePublicKey
       PARAMS ARE absent
       CERT-KEY-USAGE
           { digitalSignature, nonRepudiation, keyCertSign, cRLSign }
       PRIVATE-KEY CompositePrivateKey
   }

CompositePublicKey ::= SEQUENCE SIZE (1..MAX) OF SubjectPublicKeyInfo

CompositePrivateKey ::= SEQUENCE SIZE (1..MAX) OF OneAsymmetricKey
```

_Originally posted by @danvangeest in https://github.com/EntrustDatacard/draft-ounsworth-composite-sigs/issues/45#issuecomment-497198319_

### PR DESCRIPTION
**Pull request by danvangeest**
Separate OIDs are only necessary for keys and signature algorithms
when the signature algorithm can be combined with a pre-hash (e.g.
ecdsa-with-sha256).  This doesn't apply to composite signatures
since the pre-hash will be specific to each of the component
algorithms.

This change also uses the SIGNATURE-ALGORITHM and PUBLIC-KEY object
types of RFC 5912.

Closes #50

### CODE REVIEW

File: draft-ounsworth-pq-composite-sigs.mkd [hunk 1]
```diff
@@ -144,67 +144,107 @@ This section defines
...
+
+~~~
+    nonRepudiation;
+    digitalSignature;
+    keyCertSign; and
+    cRLSign.
+~~~
+{: artwork-name="Cert-Key-Usage-2"}
+
+_EDNOTE: BEGIN - This text is based on RFC 8410 (Ed25519 etc in X.509). It specifies a minimum required set of key usage bits to set, but doesn't preclude one from setting more.  On the other hand, the text for cms-hash-sig is below.  It is less explicit about which situations use the signature-related bits, but does preclude all other bits from being set.  Do we want to follow one or the other of these models?
```
Comment (danvangeest) [inline draft-ounsworth-pq-composite-sigs.mkd]: The text above, that is.

**Comment by ounsworth** [comment]
Make sure we close the associated Issues once this is merged.

**Comment by ounsworth** [comment]
Closes #50 
Closes #48

**Comment by danvangeest** [comment]
Closes #45

**Pull request merged**